Fix spelling errors

umlaeute opened the pull request:
- Closes: https://github.com/obsproject/obs-studio/issues/11160

### Description

Fixes a number of spelling mistakes in (log-)messages, documentation and comments.

### Motivation and Context

Proper English makes log-messages, documentation and code more readable.

### How Has This Been Tested?

still builds

### Types of changes
<!--- What types of changes does your PR introduce? Uncomment all that apply -->
<!--- - Bug fix (non-breaking change which fixes an issue) -->
<!--- - New feature (non-breaking change which adds functionality) -->
<!--- - Tweak (non-breaking change to improve existing functionality) -->
<!--- - Performance enhancement (non-breaking change which improves efficiency) -->
- Code cleanup (non-breaking change which makes code smaller or more readable)
<!--- - Breaking change (fix or feature that would cause existing functionality to change) -->
- Documentation (a change to documentation pages) 

### Checklist:
- [ ] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

umlaeute committed:
libcaption: Fix spelling errors in libcaption

comments, variablenames

umlaeute committed:
Fix typos in comments

umlaeute committed:
docs: Fix typos in documentation

umlaeute committed:
cmake: Fix typos in messages emitted by cmake

umlaeute committed:
libobs: Fix typos in variable names

umlaeute committed:
plugins: Fix typos in log-messages

norihiro reviewed (commented):
You need to follow the commit guideline.
https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst

I feel negative to see `deps/libcaption` is modified like this because the source code was carried from an external library. If it is updated or someone makes a patch to the upstream library, the merge might become difficult.

Fenrirthviti commented:
Just speaking personally, I would prefer not to bother with spellchecking comments or vendored dependencies. User facing messages and log output seems fine, assuming those log messages are our code and not from a dep.

umlaeute commented:
I was under the impression, that libcaption is unmaintained upstream and therefore is now practically maintained by the OBS project. Hence I felt safe to change it. 
In and case, it is a separate commit, so it can be left out.

Re: commit-guidelines: I followed them to the best of my knowledge, could you elaborate on what is wrong? 

Re: spell checking comments: I understand that you don't want to do the work yourself, but I fail to see why it shouldn't be approved if somebody did the work for you...

umlaeute commented on plugins/obs-outputs/librtmp/rtmp.c:
```diff
@@ -3835,7 +3835,7 @@ RTMP_ReadPacket(RTMP *r, RTMPPacket *packet)
         if (ReadN(r, (char *)&hbuf[1], 2) != 2)
         {
-            RTMP_Log(RTMP_LOGERROR, "%s, failed to read RTMP packet header 3nd byte",
+            RTMP_Log(RTMP_LOGERROR, "%s, failed to read RTMP packet header 3rd byte",
```
Well the `2nd byte` is checked a few lines above in <https://github.com/obsproject/obs-studio/blob/392f143d94e0bc37265889cf18f1b58b28f71d71/plugins/obs-outputs/librtmp/rtmp.c#L3825>

norihiro replied:
The line 3825 is reached when `m_nChannel` is 0. On the other hand, the line 3838, which you are changing, is reached when `m_nChannel` is 1. So, these are exclusive. `ReadN` at the line 3836 is reading 2nd and 3rd bytes.

umlaeute replied:
???

the full context is:
```C
        if (ReadN(r, (char *)&hbuf[1], 1) != 1)
        {
            RTMP_Log(RTMP_LOGERROR, "%s, failed to read RTMP packet header 2nd byte",
                     __FUNCTION__);
            return FALSE;
        }
```

so the error is thrown when `ReadN` fails to read one(1) additional byte (which would be the 3rd byte, as the last (2nd) byte is read in line 3812 (and an error is output if that failed=

norihiro replied:
https://github.com/obsproject/obs-studio/blob/392f143d94e0bc37265889cf18f1b58b28f71d71/plugins/obs-outputs/librtmp/rtmp.c#L3812
The byte read by above `ReadN` is the 1st byte, isn't it?

https://github.com/obsproject/obs-studio/blob/392f143d94e0bc37265889cf18f1b58b28f71d71/plugins/obs-outputs/librtmp/rtmp.c#L3836
Then, above `ReadN` should follows the 1st byte's `ReadN` and this is reading 2 bytes. So, these bytes are 2nd and 3rd bytes, aren't they?

umlaeute replied:
indeed. i was obviously confused. sorry.

in any case: this PR tries to fix spelling mistakes (like `3nd`) rather than improve the wording in general.

do you think changing `3nd`  is a regression?
do you want me to use `2nd and 3rd`? if so, within the `plugins: Fix typos in log-messages ` commit, or in a separate commit?

Fenrirthviti commented:
It isn't about doing the work, it's about touching 55 files that otherwise have no meaningful changes, impacting git history.

However, this is just my opinion.

umlaeute commented on libobs/util/simde/simde-align.h:
```diff
@@ -11,7 +11,7 @@
  *
  * This is portability layer which should help iron out some
- * differences across various compilers, as well as various verisons of
+ * differences across various compilers, as well as various versions of
```
upstream already fixed this 6 months ago:
https://github.com/simd-everywhere/simde/commit/8639feff3843dc02c8f5e2e34a2ba865d3acb546

RytoEX replied:
Regardless, it will be fixed with the unvendoring of side, and I do not like patching vendored dependencies. Please remove them.

umlaeute commented on plugins/obs-filters/rnnoise/src/kiss_fft.c:
```diff
@@ -39,7 +39,7 @@
 
 /* The guts header contains all the multiplication and addition macros that are defined for
-   complex numbers.  It also delares the kf_ internal functions.
+   complex numbers.  It also declares the kf_ internal functions.
```
upstream fixed this 4 years ago: https://github.com/xiph/rnnoise/commit/0627990e0dc0b6de607a2575b7adb0efe67c181e

RytoEX replied:
Then it will be fixed when we update or unvendor this. I do not like patching vendored dependencies and I prefer only patching them when absolutely necessary because:
1. It can cause trivial patches to fail to apply cleanly.
2. You have to check that you aren't accidentally reverting your custom changes when you update.
3. It makes it harder to verify exactly which commit you vendored or what source you used.

Please remove the changes to the rnnoise files.

umlaeute commented on UI/cmake/linux/com.obsproject.Studio.desktop:
```diff
@@ -68,7 +68,7 @@ Comment[ms_MY]=Perisian Penstriman/Rakaman Bersumber Terbuka dan Bebas
 Comment[nb_NO]=Gratis Strømming- og Opptaksprogramvare med Åpen Kildekode
 GenericName[nl_NL]=Streaming/Opname Software
-Comment[nl_NL]=Vrij en Open Source Streaming/Opname Sofware
+Comment[nl_NL]=Vrij en Open Source Streaming/Opname Software
```
thx. I submitted a fix to Crowdin

umlaeute commented on UI/xdg-data/com.obsproject.Studio.desktop:
```diff
@@ -68,7 +68,7 @@ Comment[ms_MY]=Perisian Penstriman/Rakaman Bersumber Terbuka dan Bebas
 Comment[nb_NO]=Gratis Strømming- og Opptaksprogramvare med Åpen Kildekode
 GenericName[nl_NL]=Streaming/Opname Software
-Comment[nl_NL]=Vrij en Open Source Streaming/Opname Sofware
+Comment[nl_NL]=Vrij en Open Source Streaming/Opname Software
```
thx. I submitted a fix to Crowdin

umlaeute commented:
i force-pushed with
- removed fixes for the UI (as it is managed by Crowdin) 
- removed fixes for `libnsgif` (as the current spelling is a valid variant)

i **kept** the changes to the vendored `simde` and `rnnoise` libraries, as they have been fixed upstream a while ago (several years in the case of `rnnoise`) but have yet not been synced (as of today).

umlaeute commented:
i force-pushed again with
- changes to `rrdnoise` removed
- changes to `simde` removed

luzpaz commented:
> It isn't about doing the work, it's about touching 55 files that otherwise have no meaningful changes, impacting git history.
> 
> However, this is just my opinion.

Easy enough to workaround using the `.git-blame-ignore-revs`[^1] file. Here's the doc[^2] for anyone unfamiliar.

[^1]: https://github.com/obsproject/obs-studio/blob/master/.git-blame-ignore-revs
[^2]: https://docs.github.com/en/repositories/working-with-files/using-files/viewing-and-understanding-files#ignore-commits-in-the-blame-view
